### DISCUSSION TITLE
Using a ready trained model on new input data

### DISCUSSION THREAD

**Discussion by LennardLux** [neuralhydrology/neuralhydrology#135] [Q&A] [answered]
Hi everyone,

I have a question about the application of ready trained models. When I completed the training, validation and testing process, how can I use the model to generate output for new input sequences? Is there already a function implemented for this or did I overlook it?

**Comment by gauchm** [comment, marked as answer]
If you want to run inference after training for a new set of basins and/or a different time period, the simplest way of doing so is to go to the config.yml file in the run directory, change the test period settings (`test_start_date`/`test_end_date`/`test_basin_file`), and then re-run the inference via `nh-run evaluate`. Note that this will overwrite existing test period outputs in the `test` folder of the run directory, so you might want to rename them.

If you want to do this programmatically, you can change the test period settings in your `Config` object and use the [start_evaluation](https://github.com/neuralhydrology/neuralhydrology/blob/master/neuralhydrology/evaluation/evaluate.py#L7) method.

Does that answer your question?

**Reply by LennardLux** [reply]
Yes, it does. Thanks!
